apis/iam/v1beta1/diff_test.go

```diff
@@ -0,0 +1,856 @@
+					{
+						ID:  "spec.auditConfigs[service=compute.googleapis.com]",
+						Old: "absent in desired spec",
+						New: "present in actual spec",
+					},
+				},
+			}},
+		{
+			name: "audit config log type differs",
+			desired: &IAMPolicySpec{
+				AuditConfigs: []IAMPolicyAuditConfig{
+					{
+						Service: "allServices",
+						AuditLogConfigs: []AuditLogConfig{
+							{
+								LogType: "DATA_READ",
+							},
+						},
+					},
+				},
+			},
+			actual: &IAMPolicySpec{
+				AuditConfigs: []IAMPolicyAuditConfig{
+					{
+						Service: "allServices",
+						AuditLogConfigs: []AuditLogConfig{
+							{
+								LogType: "DATA_WRITE",
+							},
+						},
+					},
+				},
+			},
+			want: &structuredreporting.Diff{
+				Fields: []structuredreporting.DiffField{
+					{
+						ID: "spec.auditConfigs[service=allServices]",
+						Old: IAMPolicyAuditConfig{
+							Service: "allServices",
+							AuditLogConfigs: []AuditLogConfig{
+								{
+									LogType: "DATA_READ",
+								},
+							},
+						},
+						New: IAMPolicyAuditConfig{
+							Service: "allServices",
+							AuditLogConfigs: []AuditLogConfig{
+								{
+									LogType: "DATA_WRITE",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "audit config exempted members differs",
+			desired: &IAMPolicySpec{
+				AuditConfigs: []IAMPolicyAuditConfig{
+					{
+						Service: "allServices",
+						AuditLogConfigs: []AuditLogConfig{
+							{
+								LogType:         "DATA_READ",
+								ExemptedMembers: []Member{"user:exempt1@example.com"},
+							},
+						},
+					},
+				},
+			},
+			actual: &IAMPolicySpec{
+				AuditConfigs: []IAMPolicyAuditConfig{
+					{
+						Service: "allServices",
+						AuditLogConfigs: []AuditLogConfig{
+							{
+								LogType:         "DATA_READ",
+								ExemptedMembers: []Member{"user:exempt2@example.com"},
+							},
+						},
+					},
+				},
+			},
+			want: &structuredreporting.Diff{
+				Fields: []structuredreporting.DiffField{
+					{
+						ID: "spec.auditConfigs[service=allServices]",
+						Old: IAMPolicyAuditConfig{
+							Service: "allServices",
+							AuditLogConfigs: []AuditLogConfig{
+								{
+									LogType:         "DATA_READ",
+									ExemptedMembers: []Member{"user:exempt1@example.com"},
+								},
+							},
+						},
+						New: IAMPolicyAuditConfig{
+							Service: "allServices",
+							AuditLogConfigs: []AuditLogConfig{
+								{
+									LogType:         "DATA_READ",
+									ExemptedMembers: []Member{"user:exempt2@example.com"},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "audit config exempted members order differs - no diff",
+			desired: &IAMPolicySpec{
+				AuditConfigs: []IAMPolicyAuditConfig{
+					{
+						Service: "allServices",
+						AuditLogConfigs: []AuditLogConfig{
+							{
+								LogType:         "DATA_READ",
+								ExemptedMembers: []Member{"user:exempt1@example.com", "user:exempt2@example.com"},
```

Comment:
The existing comment on the PR is valid. An inverse test case where the desired spec has unsorted members and the actual spec has sorted members should be added to ensure the logic is symmetrical.